packages/string-templates/src/conversion/index.js

```diff
@@ -0,0 +1,129 @@
+const { getHelperList } = require("../helpers")
+
+function getLayers(fullBlock) {
+  let layers = []
+  while (fullBlock.length) {
+    const start = fullBlock.lastIndexOf("("),
+      end = fullBlock.indexOf(")")
+    let layer
+    if (start === -1 || end === -1) {
+      layer = fullBlock.trim()
+      fullBlock = ""
+    } else {
+      const untrimmed = fullBlock.substring(start, end + 1)
+      layer = untrimmed.substring(1, untrimmed.length - 1).trim()
+      fullBlock =
+        fullBlock.slice(0, start) +
+        fullBlock.slice(start + untrimmed.length + 1, fullBlock.length)
+    }
+    layers.push(layer)
+  }
+  return layers
+}
+
+function getVariable(variableName) {
+  if (!variableName || typeof variableName !== "string") {
+    return variableName
+  }
+  // it is an array
+  const arrayOrObject = [",", "{", ":"]
+  let contains = false
+  arrayOrObject.forEach(char => {
+    if (variableName.includes(char)) {
+      contains = true
+    }
+  })
+  if (variableName.startsWith("[") && contains) {
+    return variableName
+  }
+  // it is just a number
+  if (!isNaN(parseFloat(variableName))) {
+    return variableName
+  }
+  if (variableName.startsWith("'") || variableName.startsWith('"')) {
+    return variableName
+  }
+  // extract variable
+  return `$("${variableName}")`
+}
+
+function buildList(parts, value) {
+  function build() {
+    return parts
+      .map(part => (part.startsWith("helper") ? part : getVariable(part)))
+      .join(", ")
+  }
+  if (!value) {
+    return parts.length > 1 ? `${build()}` : build()
+  } else {
+    return parts.length === 0 ? value : `${value}, ${build()}`
+  }
+}
+
+function splitBySpace(layer) {
```

Comment:
See I knew building a transpiler was completely trivial and not complicated at all. This file proves it once and for all!